mautic/mautic
# Improve Campaign Trigger command Scheduled Events Queries

⚠️ **REQUIRES PR 6247 (https://github.com/mautic/mautic/pull/6247) TO BE MERGED/APPLIED FIRST** ⚠️ 

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
**REQUIRES PR 6247 (https://github.com/mautic/mautic/pull/6247)**

Campaigns with large ingestion (~100k + daily) create a bottlenecks when processing scheduled events. The existing query for GetScheduledCounts relied on joins and aggregations with large cardinality. This change hopes to mitigate that with no backwards breaking issues. Also avoids doing Counts if the command is run in --quiet mode (which should be the production use case to minimize log noise).

#### Steps to reproduce the bug:
1. 
2. 

#### Steps to test this PR:
1. push leads into a campaign that has a delayed event
2. run command mautic:campaign:trigger -i [x] where x is the affected campaign.
3. Check that any scheduled events ran correctly
4. repeat steps 1-3 with command mautic:campaign:trigger -i [x] --quiet

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
